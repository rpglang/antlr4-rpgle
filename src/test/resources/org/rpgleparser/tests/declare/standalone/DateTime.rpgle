        DCL-S duedate DATE;
        DCL-S displayDate DATE(*YMD);
        DCL-S startTime TIME;
        DCL-S displayTime TIME(*USA);
        DCL-S start TIMESTAMP;