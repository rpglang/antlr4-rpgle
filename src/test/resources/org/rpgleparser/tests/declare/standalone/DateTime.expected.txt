/*===TOKENS===*/
DS_Standalone            <DCL-S>
ID                       <duedate>
DATE_                    <DATE>
FREE_SEMI                <;>
DS_Standalone            <DCL-S>
ID                       <displayDate>
DATE_                    <DATE>
'('                      <(>
SPLAT_YMD                <*YMD>
')'                      <)>
FREE_SEMI                <;>
DS_Standalone            <DCL-S>
ID                       <startTime>
TIME                     <TIME>
FREE_SEMI                <;>
DS_Standalone            <DCL-S>
ID                       <displayTime>
TIME                     <TIME>
'('                      <(>
SPLAT_USA                <*USA>
')'                      <)>
FREE_SEMI                <;>
DS_Standalone            <DCL-S>
ID                       <start>
TIMESTAMP                <TIMESTAMP>
FREE_SEMI                <;>
/*===TREE===*/
(r
  (statement
    (dspec
      DCL-S
      (identifier (free_identifier (idOrKeyword duedate)))
      (datatype (datatypeName DATE))
      ;
    )
  )
  (statement
    (dspec
      DCL-S
      (identifier (free_identifier (idOrKeyword displayDate)))
      (datatype
        (datatypeName DATE)
        (args ( (expression (identifier (all (symbolicConstants *YMD)))) ))
      )
      ;
    )
  )
  (statement
    (dspec
      DCL-S
      (identifier (free_identifier (idOrKeyword startTime)))
      (datatype (datatypeName TIME))
      ;
    )
  )
  (statement
    (dspec
      DCL-S
      (identifier (free_identifier (idOrKeyword displayTime)))
      (datatype
        (datatypeName TIME)
        (args ( (expression (identifier (all (symbolicConstants *USA)))) ))
      )
      ;
    )
  )
  (statement
    (dspec
      DCL-S
      (identifier (free_identifier (idOrKeyword start)))
      (datatype (datatypeName TIMESTAMP))
      ;
    )
  )
)
/*======*/